Allow creating ignition-gazebo environments in python

diegoferigo opened the pull request:
This PR is quite big but it could't be split in smaller chunks since all the modifications are related to each other.

### C++

- A new plugin `gympp::plugins::RobotController` wraps the `gympp::gazebo::IgnitionRobot` class. It can be attached to any sdf model and registers a `gympp::Robot` interface in the `gympp::gazebo::RobotSingleton`.
- The logic to create a gazebo server and gui has been moved from `IgnitionEnvironment` to a new class `gympp::gazebo::GazeboWrapper`.
- Implemented remaining joint-related method in `gympp::gazebo::IgnitionRobot`.

### Python

- `gympp::gazebo::RobotSingleton` and `gympp::gazebo::GazeboWrapper` are now binded with swig.
- New `IgnitionPythonEnv` class that wraps `gympp::gazebo::GazeboWrapper` and provides the machinery to create gym environments that interface with ignition gazebo directly in python.
- New `CartPoleIgnitionPython-v0` environment. It matches as much as possible the upstream `CartPole-v1`, but the simulation runs inside ignition gazebo.

Closes #14

traversaro commented on bindings/gympp.i:
```diff
@@ -5,15 +5,19 @@
 #include "gympp/Metadata.h"
 #include "gympp/Robot.h"
 #include "gympp/Space.h"
+#include <stdint.h>
```
Nitpick: for a C++ file, the recommended include is `cstdint`(see https://stackoverflow.com/questions/13642827/cstdint-vs-stdint-h). Or perhaps you did this for some weird SWIG incompatibility.

diegoferigo replied:
I read https://github.com/swig/swig/issues/495 and I forgot to use the C++ styled inclusion. I did a test and using `cstdint` in the include list works, but `stdint.h` should be passed to the swig's `%include` directive in order to find the file.

traversaro commented:
I have a few comments that came to my mind by reading this code, even if they are not directly related to this PR: 
* It is possible to set three different timestep for the agent update, the physics fixed time step and for the PID update rate? While this may seem an advanced feature, it is quite a basic requirement for modelling a real robot, in which you decrease the physics fixed time step until the numerical error is small  enough, but the PID timestep and the "Agent" are fixed by architectural choices on the real robot (typical value may be 1 Khz for the low-level  PID, and 100 Hz for the higher level agent, if it is directly sending values to the low-level PIDs). See https://github.com/robotology/gazebo-yarp-plugins/issues/69 for an old issue on this in gazebo-yarp-plugins . 
* Are you taking into account the possibility that a model has 0-dof (i.e. `fixed`) joints? The iCub model has some of those to model six-axis F/T torque sensors, see the "Modeling a Real Force/Torque Sensor" in http://gazebosim.org/tutorials?tut=force_torque_sensor&cat=sensors .

diegoferigo commented:
> It is possible to set three different timestep for the agent update, the physics fixed time step and for the PID update rate?

I agree and it is a feature that we have to have. Currently this is not supported but I don't see any mayor blocking factors to implement it. I played a bit with PID tuning on the CartPole environment last week, and definitely having the possibility to chose different rates for the robot components would greatly simply the task.

> Are you taking into account the possibility that a model has 0-dof (i.e. fixed) joints?

Fixed joints are currently included in the returned list of joint names but their position cannot be read nor set. Without referring to complex models like iCub, the CartPole model has a fixed joint from the base to the rail. It currently raises a warning (you got it without running the code, good catch ;) ), I am aware of it but I still didn't tackled it.
